src/gui/optionsdialog.cpp

```diff
@@ -176,6 +192,20 @@ OptionsDialog::OptionsDialog(QWidget *parent)
     connect(m_ui->buttonBox, &QDialogButtonBox::clicked, this, &OptionsDialog::applySettings);
     // Languages supported
     initializeLanguageCombo();
+    initializeThemeCombo();
+    connect(m_ui->comboTheme, &QComboBox::currentTextChanged, [&](const QString &newText)
+    {
+        if (newText == tr("Select...")) {
+            m_qssFilePath = QFileDialog::getOpenFileName(
+                this, tr("Select Qss File"), "", tr("Style File (*.qss)"));
+            m_ui->comboTheme->clear();
+            initializeThemeCombo(); // don't leave currentText as `Select...`
```

Comment:
I don't see the need to reinitialize it.
I do not understand why you chose a text-based methods to manage it. Since you have a predetermined number of items, you know their indexes, so it is better to use more efficient index-based methods. Besides, all these functions (fromThemeStr, themeStr) are unnecessary.
```c++
m_ui->comboTheme->addItem(tr("Default"));
if (UIThemeManager::instance()->isCustomAvailable())
    m_ui->comboTheme->addItem(tr("Custom"));
m_ui->comboTheme->insertSeparator(m_ui->comboTheme->count());
m_ui->comboTheme->addItem(tr("Select..."));
m_ui->comboTheme->setCurrentIndex(UIThemeManager::instance()->theme() == UIThemeManager::Theme::Default ? 0 : 1);

connect(m_ui->comboTheme, qOverload<int>(&QComboBox::currentIndexChanged), [&](int index)
{
    if (index == (m_ui->comboTheme->count() - 1)) {
        m_qssFilePath = QFileDialog::getOpenFileName(this, tr("Select Qss File"), "", tr("Style File (*.qss)"));
        if (!m_qssFilePath.isEmpty()) {
            m_ui->comboTheme->blockSignals(true);
            if (m_ui->comboTheme->count() == 3)
                m_ui->comboTheme->insertItem(1, tr("Custom"));
            m_ui->comboTheme->setCurrentIndex(1);
            m_ui->comboTheme->blockSignals(false);
        }
    }
});
```

Reply:
thanks, didn't know about `blockSignals`, still learning XD.